x-pack/legacy/plugins/code/server/git_operations.ts

```diff
@@ -382,38 +222,46 @@ export class GitOperations {
         path: '',
         type: FileTreeItemType.Directory,
       };
-      const tree = await GitOperations.readTree(gitdir, commit.treeId);
-      await this.fillChildren(root, tree, { skip, limit, flatten });
+      const tree = await this.readTree(git, commit.treeId);
+      await this.fillChildren(git, root, tree, { skip, limit, flatten });
       if (path) {
-        await this.resolvePath(root, tree, path.split('/'), { skip, limit, flatten });
+        await this.resolvePath(git, root, tree, path.split('/'), { skip, limit, flatten });
       }
       return root;
     } else {
-      const obj = await GitPrime.readObject({ gitdir, oid: commit.id, filepath: path });
-      const result: FileTree = {
-        name: path.split('/').pop() || '',
-        path,
-        type: type2item(obj.type!),
-        sha1: obj.oid,
-      };
-      if (result.type === FileTreeItemType.Directory) {
-        await this.fillChildren(
-          result,
-          {
-            gitdir,
-            entries: (obj.object as TreeDescription).entries,
-            oid: obj.oid,
-          },
-          { skip, limit, flatten }
-        );
+      if (path) {
+        const file = (await this.readTree(git, commit.id, path)).entries[0];
+        const result: FileTree = {
+          name: path.split('/').pop() || '',
+          path,
+          type: type2item(file.type!),
+          sha1: file.id,
+        };
+        if (file.type === 'tree') {
```

Comment:
nit: for this `tree` and `blob`, it might be easier to track in the future if we can add an Enum.